cmd/main.go

```diff
@@ -17,34 +16,39 @@ import (
 var splitCmd = &cli.Command{
 	Name: "split",
 	Action: func(c *cli.Context) error {
-		if !c.Args().Present() {
-			return fmt.Errorf("must pass a CAR file to split")
-		}
-		path := c.Args().First()
-		dir := filepath.Dir(path)
-		name := strings.TrimRight(filepath.Base(path), ".car")
-
 		var strategy carbites.Strategy
 		if c.String("strategy") == "treewalk" {
 			strategy = carbites.Treewalk
 		}
+
 		size := c.Int("size")
 		fmt.Printf("Splitting into ~%d byte chunks using strategy \"%s\"\n", size, c.String("strategy"))
 
-		var spltr carbites.Splitter
+		var path, dir, name string
+		var fi io.Reader
 		var err error
-		if strategy == carbites.Treewalk {
-			// does not cache in memory
-			spltr, err = carbites.NewTreewalkSplitterFromPath(path, size)
```

Comment:
Can we please keep this? It allows splitting with the treewalk strategy without buffering the entire CAR in memory.

Reply:
Yes, sorry. Kept it now.